### DISCUSSION TITLE
How to fix connected link position when I move the Link

### DISCUSSION THREAD

**Discussion by reshin-pulapaddy** [clientIO/joint#2240] [Q&A]
### Introduction

Here The horizontal link was connected by other links to two rectangles
When I move the source, the other Link-connected points are also changed, but when I move the target it works correctly.
I need to fix the other link when I move the source
how to do that?
![link](https://github.com/clientIO/joint/assets/117301232/c1d10694-866e-4820-8635-f98a24eeca69)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
It depends on how you define the link's anchor.
- **connectionLength** - the anchor is at a specific distance from the `source` (it's your case - the distance does not change when you move the `target`)
- **connectionRatio** - the ratio of the link's length (e.g. at 30% of length)

See the [link anchors](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#linkAnchors) docs.

**Reply by reshin-pulapaddy** [reply]
while creating the paper I am not passing any defaultLinkAnchor,
I am connecting the link through UI only.
can you please provide any example of that

const paper = (this.paper = new joint.dia.Paper({
			width: this.backgroundImage.width,
			height: this.backgroundImage.height,
			gridSize: 5,
			drawGrid: true,
			model: graph,
			snapLinksSelf: true,
			restrictTranslate: true,
			cellViewNamespace: joint.shapes,
			autoResizePaper: false,
			defaultConnectionPoint: { name: 'boundary' },
			defaultLink: new joint.shapes.standard.Link({
				attrs: {
					line: {
						stroke: colorPalette.Grey,
						targetMarker: {
							type: 'path',
							d: 'M 0 0 0 0',
						},
					},
				},
			}),
			interactive: this.interactive,
			async: true,
			snapLabels: true,
			sorting: joint.dia.Paper.sorting.APPROX,
			connectionStrategy: joint.connectionStrategies.pinAbsolute,
			validateConnection: () => {
				return true;
			},
		}));

**Reply by kumilingus** [reply]
Please create a JSFiddle with exactly what is shown in the video and we can take a look at it.

---

[Here](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) is also a guide on how to use the markdown language on GitHub.

**Reply by reshin-pulapaddy** [reply]
https://jsfiddle.net/reshin_pulapaddy/bkw56rea/53/

**Reply by kumilingus** [reply]
- Connection Perpendicular (via `paper.options.defaultLinkAnchor`)
https://jsfiddle.net/kumilingus/g7jtLpvf/

- Connection Ratio (via `paper.options.connectionStrategy = connectionStrategies.pinRelative`).
https://jsfiddle.net/kumilingus/L4x3dagb/
